pkg/vsphere/datastore/datastore.go

```diff
@@ -319,32 +319,28 @@ func (d *Helper) mkRootDir(ctx context.Context, rootdir string) error {
 		log.Infof("datastore root %s already exists", rooturl)
 	}
 
-	d.RootURL = rooturl
+	d.RootURL.FromString(rooturl)
 	return nil
 }
 
-// Return the root of the datastore path (without the [datastore] portion)
-func (d *Helper) rootDir() string {
-	return strings.SplitN(d.RootURL, " ", 2)[1]
-}
-
-// Parse the datastore format ([datastore1] /path/to/thing) to groups.
-var datastoreFormat = regexp.MustCompile(`^\[([\w\d\(\)-_\.\s]+)\]`)
-var pathFormat = regexp.MustCompile(`\s([\/\w-_\.]*$)`)
-
 // Converts `[datastore] /path` to ds:// URL
 func ToURL(ds string) (*url.URL, error) {
```

Comment:
If we're going to do this we should allow symmetry - so that we can parse back to DatastorePath from the url format.
Two options:
1. add a FromURL field that parses the url, treating the Host part as the datastore - this cannot be with url.Parse because that doesn't handle many characters in the host portion.
2. update FromString so that it can accept an "ipv6" encoded form: `ds://[datastore name with whatever]/path/to/other`